app/models/post.rb

```diff
@@ -1,3 +1,17 @@
 class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User'
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments
+  has_many :likes
+
+  after_create :update_user_post_counter
+
```

Comment:
- [ ] Instead of calling `after_create` callback on your `update_user_post_counter` I would suggest you to use `after_save` callback because `after_create` callback is invoked only once, immediately after the record is first created and does not get triggered when updating an existing record. On the other hand `after_save` callback is called every time you save the object, regardless of whether it’s a new or existing record.